bitcoin/src/string.rs

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Bitcoin string parsing utilities.
+//!
+//! This module provides utility types and traits
+//! to support handling and parsing strings within `rust-bitcoin`.
+
+use crate::prelude::String;
+
+/// Hex parsing error
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[non_exhaustive]
+pub enum FromHexError<E> {
+    /// The input was not a valid hex string, contains the error that occurred while parsing.
+    ParseHex(E),
+    /// The input is missing `0x` prefix, contains the invalid input.
+    MissingPrefix(String),
+}
+
+/// Trait that allows types to be initialized from hex strings
+pub trait FromHexStr {
+    /// An error occurred while parsing the hex string.
+    type Error;
+
+    /// Parses provided string as hex requiring 0x prefix.
+    ///
+    /// This is intended for user-supplied inputs or already-existing protocols in which 0x prefix is used.
+    fn from_hex_str<S: AsRef<str> + Into<String>>(s: S) -> Result<Self, FromHexError<Self::Error>>
+    where
+        Self: Sized,
```

Comment:
Oh, there should be `: Sized` bound on the trait, not here. The compiler didn't suggest the right thing in this case. Sorry for only noticing now :(

Reply:
It's ok, better we find it now than later